readme/MacBuild.md

```diff
@@ -1,141 +1,264 @@
+jhbuild build pygments
+
+# Now we can build all dependencies.
+# Warning: the following steps compile a lot of code and take a while to run.
+jhbuild bootstrap-gtk-osx
+jhbuild build meta-gtk-osx-gtk3
+jhbuild build gtksourceview3
 ```
-#### Potential Errors
-##### Unknown module (macOS Mojave 10.14)
 
-If there is an error like:
+### Step 4: Clone the Xournal++ code
+
+If you haven't already, you'll need to clone the Xournal++ code in order to
+compile it and its dependencies from source:
+
 ```sh
-xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory'/Library/Developer/CommandLineTools' is a command line tools instance
+git clone http://github.com/xournalpp/xournalpp.git
 ```
 
-Follow these steps to resolve the issue:
-1. Install Xcode (get it from [here](https://developer.apple.com/xcode/)) if you don't have it yet.
-2. Accept the Terms and Conditions.
-3. Ensure Xcode app is in the /Applications directory (**NOT** /Users/{user}/Applications).
-4. Point xcode-select to the Xcode app Developer directory using the following command:
-5. sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
-6. Make sure your Xcode app path is correct.
-   1. Xcode: /Applications/Xcode.app/Contents/Developer
-   2. Xcode-beta: /Applications/Xcode-beta.app/Contents/Developer
-   
-Steps are from this [source](https://stackoverflow.com/questions/17980759/xcode-select-active-developer-directory-error/17980786#17980786). A big thanks to tjmetha and Rob Bednark!
+### Step 5: Build required dependencies not in gtk-osx
+
+Some other dependencies are not provided by gtk-osx.
+We provide a jhbuild moduleset that defines how these dependencies will be
+downloaded and built.
+
+To build them, run the following command in the `mac-setup` folder of the
+`xournalpp` repository you just cloned:
 
-### 6. Build sndfile
-**Recording and playing audio is not yet supported for MacOS.**
 ```sh
-./build-sndfile.sh
+jhbuild -m xournalpp.modules build meta-xournalpp-deps
 ```
 
-### 7. Build adwaita icon theme
+**Note**: recording and playing audio is not yet supported for MacOS.
+
+### Last step: build Xournal++ and package it as .app
+
+From the `mac-setup` directory, this can be done completely automatically using:
 ```sh
-jhbuild build adwaita-icon-theme
+./complete-build.sh "$HOME"/gtk
 ```
 
-### 8. Build GtkSourceView
+Or manually with:
 ```sh
-jhbuild build gtksourceview3
+# from the top-level folder
+
+# build and install to local gtk prefix
+mkdir -p build && cd build
+cmake .. -GNinja -DCMAKE_INSTALL_PREFIX:PATH="$HOME"/gtk/inst
+cmake --build .
+cmake --install .
+cd ..
+
+# build application
+cd mac-setup
+jhbuild run bash ./build-app.sh "$HOME"/gtk
 ```
 
-## Build Xournal++ and package it as .app
-
-### Automated step
+You also run the application from the `build` folder without installing:
 
 ```sh
-./complete-build.sh $HOME/gtk
+cd build
+jhbuild run ./xournalpp
```

Comment:
I don't think that works as of now (c.f. #4153). What you can do is run the installed version using `jhbuild run xournalpp` without creating the .app-bundle.

Reply:
Actually, I can't get this to run with the special build user. I'll need to see if this works with the binary blob.

Reply:
It doesn't work with the binary blob, unfortunately.